### PR TITLE
[Keyboard] Add mokey12x2 keyboard

### PR DESCRIPTION
**Pull request by rhmokey**
## Description
Add mokey12x2 keyboard

## Types of Changes

- [x] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [x] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/mokey/mokey12x2/mokey12x2.c [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+ */
+
+#include "quantum.h"
+
+#include "quantum.h"
+
+#if defined(ENCODER_ENABLE)
+
+bool encoder_update_kb(uint8_t index, bool clockwise) {
+    if (!encoder_update_kb(index, clockwise)) {
```
Comment (zvecr) [inline keyboards/mokey/mokey12x2/mokey12x2.c:24]: Clearly not tested, as this produces an infinite loop.
```suggestion
    if (!encoder_update_user(index, clockwise)) {
```

File: keyboards/mokey/mokey12x2/halconf.h [hunk 2]
```diff
@@ -0,0 +1,21 @@
+/* Copyright 2020 QMK
```
Comment (zvecr) [inline keyboards/mokey/mokey12x2/halconf.h:1]: Nothing is configured for SPI, this file should be removed.

File: keyboards/mokey/mokey12x2/info.json [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+    "maintainer": "qmk",
+    "usb": {
+        "vid": "0x6653",
+        "pid": "0x012A",
+        "device_version": "0.0.1"
+    },
+    "diode_direction": "COL2ROW",
+    "matrix_pins": {
+        "cols": ["A6", "A5", "A4", "A3", "B9"],
+        "rows": ["A2", "A1", "A0", "NO_PIN"]
```
Comment (zvecr) [inline keyboards/mokey/mokey12x2/info.json:14]: Padding with `NO_PIN` here is incorrect, given nothing will then trigger the matrix locations.

**Review by zvecr — CHANGES_REQUESTED** [review]
Also, you should not close and reopen PRs. Push updates to the same branch and they will be reflected here.

File: keyboards/mokey/mokey12x2/mokey12x2.c [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "quantum.h"
+
+#include "quantum.h"
```
Comment (zvecr) [inline keyboards/mokey/mokey12x2/mokey12x2.c:19]:
```suggestion
#include "quantum.h"
```

File: keyboards/mokey/mokey12x2/keymaps/default/keymap.c [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+        KC_CAPS, KC_A,    KC_S,    KC_D,
+        KC_LSFT, KC_Z,    KC_X,    KC_C
+    ),
+
+    [1] = LAYOUT(
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS
+    )
```
Comment (zvecr) [inline keyboards/mokey/mokey12x2/keymaps/default/keymap.c:31]:
```suggestion
    )
```

File: keyboards/mokey/mokey12x2/keymaps/via/keymap.c [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+    ),
+
+    [3] = LAYOUT(
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS
+    )
```
Comment (zvecr) [inline keyboards/mokey/mokey12x2/keymaps/via/keymap.c:45]:
```suggestion
	)
```

File: keyboards/mokey/mokey12x2/rules.mk [hunk 7]
```diff
@@ -0,0 +1,16 @@
...
+MOUSEKEY_ENABLE = yes	# Mouse keys
+EXTRAKEY_ENABLE = yes	# Audio control and System control
+CONSOLE_ENABLE = no	# Console for debug
+COMMAND_ENABLE = no    # Commands for debug and configuration
+NKRO_ENABLE = yes           # Enable N-Key Rollover
+BACKLIGHT_ENABLE = no
+RGBLIGHT_ENABLE = yes
+SLEEP_LED_ENABLE = no
+ENCODER_ENABLE = yes
+
```
Comment (zvecr) [inline keyboards/mokey/mokey12x2/rules.mk:14]:
```suggestion
```

File: keyboards/mokey/mokey12x2/info.json [hunk 8]
```diff
@@ -0,0 +1,56 @@
...
+        "pin": "B1"
+    },
+    "encoder": {
+        "rotary": [
+            {"pin_a": "B12", "pin_b": "B13"},
+            {"pin_a": "B14", "pin_b": "B15"}
+        ]
+    },
+    "processor": "STM32F103",
+    "bootloader": "stm32duino",
```
Comment (zvecr) [inline keyboards/mokey/mokey12x2/info.json:26]:
```suggestion
    "bootloader": "stm32duino",
    "features": {
        "bootmagic": true,
        "command": false,
        "console": false,
        "extrakey": true,
        "mousekey": true,
        "nkro": true,
        "rgblight": true,
        "encoder": true
    },
```

File: keyboards/mokey/mokey12x2/config.h [hunk 9]
```diff
@@ -0,0 +1,31 @@
+/*
```
Comment (zvecr) [inline keyboards/mokey/mokey12x2/config.h:1]: This file should be deleted

File: keyboards/mokey/mokey12x2/info.json [hunk 10]
```diff
@@ -0,0 +1,56 @@
...
+        "pid": "0x012A",
+        "device_version": "0.0.1"
+    },
+    "diode_direction": "COL2ROW",
+    "matrix_pins": {
+        "cols": ["A6", "A5", "A4", "A3", "B9"],
+        "rows": ["A2", "A1", "A0", "A7"]
+    },
+    "rgblight": {
+        "pin": "B1"
```
Comment (zvecr) [inline keyboards/mokey/mokey12x2/info.json:17]:
```suggestion
        "pin": "B1",
        "led_count": 19,
        "animations": {
            "alternating": true,
            "breathing": true,
            "christmas": true,
            "knight": true,
            "rainbow_mood": true,
            "rainbow_swirl": true,
            "rgb_test": true,
            "snake": true,
            "static_gradient": true,
            "twinkle": true
        }
```

**Review by zvecr — CHANGES_REQUESTED** [review]

**Review by zvecr — CHANGES_REQUESTED** [review]
Also,
* readme.md does not match [template](https://github.com/qmk/qmk_firmware/blob/master/data/templates/keyboard/readme.md)
* licence headers are inconsistent with attribution

File: keyboards/mokey/mokey12x2/mokey12x2.c [hunk 11]
```diff
@@ -0,0 +1,41 @@
...
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "quantum.h"
+
+#if defined(ENCODER_ENABLE)
+
+bool encoder_update_user(uint8_t index, bool clockwise) {
+    if (!encoder_update_user(index, clockwise)) {
```
Comment (zvecr) [inline keyboards/mokey/mokey12x2/mokey12x2.c:22]: Please pay more attention when applying these suggestions, you were only asked to change one line and have reintroduced the original issue in a worse way....
```suggestion
bool encoder_update_kb(uint8_t index, bool clockwise) {
    if (!encoder_update_user(index, clockwise)) {
```

**Review by zvecr — CHANGES_REQUESTED** [review]

File: keyboards/mokey/mokey12x2/mokey12x2.c (hunk 11 above)
Comment (rhmokey) [reply keyboards/mokey/mokey12x2/mokey12x2.c]: Sorry

File: keyboards/mokey/mokey12x2/mokey12x2.c [hunk 12]
```diff
@@ -0,0 +1,39 @@
...
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include "quantum.h"
+
+#if defined(ENCODER_ENABLE)
+
+bool encoder_update_kb(uint8_t index, bool clockwise) {
+    if (!encoder_update_kb(index, clockwise)) {
```
Comment (zvecr) [inline keyboards/mokey/mokey12x2/mokey12x2.c:20]: FFS, for the last time, apply the suggestion correctly

```suggestion
bool encoder_update_kb(uint8_t index, bool clockwise) {
    if (!encoder_update_user(index, clockwise)) {
```

File: keyboards/mokey/mokey12x2/mokey12x2.c (hunk 11 above)
Comment (rhmokey) [reply keyboards/mokey/mokey12x2/mokey12x2.c]: Have been changed

**Review by drashna — APPROVED** [review]

**Comment by rhmokey** [comment]
Wish to merge

**Review by drashna — APPROVED** [review]

**Comment by rhmokey** [comment]
Hello, may I ask that there is no problem with my submission, why is it not merged? Can't qmk add keyboard firmware now?

File: keyboards/mokey/mokey12x2/readme.md [hunk 13]
```diff
@@ -0,0 +1,26 @@
+# mokey12x2
+
+*mokey12x2*
```
Comment (fauxpark) [inline keyboards/mokey/mokey12x2/readme.md:3]:
```suggestion
```

File: keyboards/mokey/mokey12x2/info.json [hunk 14]
```diff
@@ -0,0 +1,79 @@
...
+        "extrakey": true,
+        "mousekey": true,
+        "nkro": true,
+        "rgblight": true,
+        "encoder": true
+    },
+    "layouts": {
+        "LAYOUT": {
+            "layout": [
+                {"matrix": [0, 0], "x":0.25, "y":2.75},
```
Comment (fauxpark) [inline keyboards/mokey/mokey12x2/info.json:53]: The keys in this layout are not in the right order. The keymaps may also need to be adjusted.
image

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/mokey/mokey12x2/keymaps/via/rules.mk [hunk 15]
```diff
@@ -0,0 +1,2 @@
+VIA_ENABLE = yes
+LTO_ENABLE = yes
```
Comment (sigprof) [inline keyboards/mokey/mokey12x2/keymaps/via/rules.mk:2]: You need to add `ENCODER_MAP_ENABLE = yes` to `rules.mk`, and then add the [encoder map](https://docs.qmk.fm/#/feature_encoders?id=encoder-map) to `keymap.c`; then adjust the VIA JSON that you should be submitting to [the-via/keyboards](https://github.com/the-via/keyboards/) according to the [VIA docs on encoder support](https://www.caniusevia.com/docs/layouts#rotary-encoders).  Do the same change for the `default` keymap too (encoder map is now considered to be the preferred way to configure encoders).

However, do not set `ENCODER_MAP_ENABLE = yes` at the keyboard level — this option is allowed only at the keymap level.

**Review by fauxpark — COMMENTED** [review]
Are the keys at the top meant to be encoders? They should not be part of the keymap; use the [Encoder Map](https://docs.qmk.fm/#/feature_encoders?id=encoder-map) feature instead.

**Review by sigprof — CHANGES_REQUESTED** [review]
A keyboard with encoders should use encoder map at least in the `via` keymap, so that the encoder rotation actions could be configured in VIA.

File: keyboards/mokey/mokey12x2/readme.md [hunk 16]
```diff
@@ -0,0 +1,12 @@
+# mokey12x2
+
+A Blue Pill STM32F103C8T6-based keyboard.
+
+Keyboard Maintainer: [rhmokey](https://github.com/rhmokey)  
+Hardware Supported: Blue Pill STM32F103C8T6
```
Comment (fauxpark) [inline keyboards/mokey/mokey12x2/readme.md:6]:
```suggestion
* Keyboard Maintainer: [rhmokey](https://github.com/rhmokey)
* Hardware Supported: Blue Pill STM32F103C8T6
```

File: keyboards/mokey/mokey12x2/mokey12x2.c [hunk 17]
```diff
@@ -0,0 +1,18 @@
+/* Copyright 2023 Mokey
```
Comment (fauxpark) [inline keyboards/mokey/mokey12x2/mokey12x2.c:1]: This file can be deleted.

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Comment by rhmokey** [comment]
Why is it not merged?

**Pull request closed without merging**